packages/amplify-graphql-relational-transformer/src/resolver/ddb-references-generator.ts

```diff
@@ -165,8 +165,116 @@ export class DDBRelationalReferencesResolverGenerator extends DDBRelationalResol
+      totalExpressionNames['#sortKeyName'] = str(sortKeyName);
+      totalExpressionValues[':sortKeyName'] = this.buildKeyValueExpression(primaryKeyFields[1], ctx.output.getObject(object.name.value)!);
+    }
+
+    const resolverResourceId = ResolverResourceIDs.ResolverResourceID(object.name.value, field.name.value);
+    const resolver = ctx.resolvers.generateQueryResolver(
+      object.name.value,
+      field.name.value,
+      resolverResourceId,
+      dataSource as any,
+      MappingTemplate.s3MappingTemplateFromString(
+        print(
+          compoundExpression([
+            iff(ref('ctx.stash.deniedField'), raw('#return($util.toJson(null))')),
+            set(
+              ref(PARTITION_KEY_VALUE),
+              methodCall(
+                ref('util.defaultIfNull'),
+                ref(`ctx.stash.connectionAttibutes.get("${primaryKeyFields[0]}")`),
+                ref(`ctx.source.${primaryKeyFields[0]}`),
+              ),
+            ),
+            ifElse(
+              or([
+                methodCall(ref('util.isNull'), ref(PARTITION_KEY_VALUE)),
+                ...primaryKeyFields.slice(1).map((f) => raw(`$util.isNull($ctx.source.${f})`)),
+              ]),
+              raw('#return'),
+              compoundExpression([
+                set(ref('GetRequest'), obj({ version: str('2018-05-29'), operation: str('Query'), index: str(indexName) })),
+                qref(
+                  methodCall(
+                    ref('GetRequest.put'),
+                    str('query'),
+                    obj({
+                      expression: str(totalExpressions.join(' AND ')),
+                      expressionNames: obj(totalExpressionNames),
+                      expressionValues: obj(totalExpressionValues),
+                    }),
+                  ),
+                ),
+                iff(
+                  not(isNullOrEmpty(authFilter)),
+                  qref(
+                    methodCall(
+                      ref('GetRequest.put'),
+                      str('filter'),
+                      methodCall(ref('util.parseJson'), methodCall(ref('util.transform.toDynamoDBFilterExpression'), authFilter)),
+                    ),
+                  ),
+                ),
+                toJson(ref('GetRequest')),
+              ]),
+            ),
+          ]),
+        ),
+        `${object.name.value}.${field.name.value}.req.vtl`,
+      ),
+      MappingTemplate.s3MappingTemplateFromString(
+        print(
+          DynamoDBMappingTemplate.dynamoDBResponse(
+            false,
+            ifElse(
+              and([not(ref('ctx.result.items.isEmpty()')), equals(ref('ctx.result.scannedCount'), int(1))]),
+              toJson(ref('ctx.result.items[0]')),
+              compoundExpression([
+                iff(and([ref('ctx.result.items.isEmpty()'), equals(ref('ctx.result.scannedCount'), int(1))]), ref('util.unauthorized()')),
+                toJson(nul()),
```

Comment:
For hasOne with reference, it could return the result `scannedCount` greater than `1` (as there might be multiple items by querying the index if more than two children with same parent reference id). Are we still returning `null` for this case in the response? Or is there a plan to prevent the child record to be created? For either case, it is better to have TODO comment here and address it in separate PR